src/migrate/Migrator.js

```diff
@@ -403,11 +405,17 @@ export function getMergedConfig(config, currentConfig) {
   // default to fs migrations to maintain compatibility
   if (!mergedConfig.migrationSource) {
     mergedConfig.migrationSource = new FsMigrations(
-      mergedConfig.directory,
-      mergedConfig.sortDirsSeparately
+      mergedConfig.globPatterns,
+      mergedConfig.sortDirsSeparately,
+      mergedConfig.globOptions
     );
   }
 
+  if (!mergedConfig.directory) {
+    const directory = await mergedConfig.migrationSource.getMigrationDirs();
```

Comment:
This is another breaking change. What would I return for say the `WebpackMigrationSource` which doesn't have directories?